starknet-rpc-test/call.rs

```diff
@@ -165,36 +161,33 @@ async fn works_on_mutable_call_without_modifying_storage(#[future] madara: Madar
     let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS);
 
     let contract_artifact: SierraClass = serde_json::from_reader(
-        std::fs::File::open(env!("CARGO_MANIFEST_DIR").to_owned() + "/contracts/HelloStarknet.sierra.json").unwrap(),
+        std::fs::File::open(env!("CARGO_MANIFEST_DIR").to_owned() + "/contracts/Counter.sierra.json").unwrap(),
     )
     .unwrap();
 
-    let declaration = account.declare(
+    let declare_tx = account.declare(
         contract_artifact.clone().flatten().unwrap().into(),
-        FieldElement::from_hex_be("0xdf4d3042eec107abe704619f13d92bbe01a58029311b7a1886b23dcbb4ea87").unwrap(), // compiled class hash
+        FieldElement::from_hex_be("0x0785fa5f2bacf0bfe3bc413be5820a61e1ea63f2ec27ef00331ee9f46ad07603").unwrap(), // compiled class hash
     );
     let contract_factory = ContractFactory::new(contract_artifact.class_hash().unwrap(), account.clone());
 
-    let mut deployment = contract_factory.deploy(vec![], FieldElement::ZERO, true);
+    let mut deploy_tx = contract_factory.deploy(vec![], FieldElement::ZERO, true);
 
     // manually setting fee else estimate_fee will be called and it will fail
     // as contract is not declared yet (declared in the same block as deployment)
-    deployment = deployment.max_fee(FieldElement::from_hex_be("0x1000000000").unwrap());
+    deploy_tx = deploy_tx.max_fee(FieldElement::from_hex_be("0x1000000000").unwrap());
 
     // manually incrementing nonce else as both declare and deploy are in the same block
     // so automatic nonce calculation will fail
     let nonce = rpc.get_nonce(BlockId::Tag(BlockTag::Latest), account.address()).await.unwrap();
-    deployment = deployment.nonce(nonce + FieldElement::ONE);
+    deploy_tx = deploy_tx.nonce(nonce + FieldElement::ONE);
```

Comment:
Can't you chain those:

```rust
let nonce = rpc.get_nonce(BlockId::Tag(BlockTag::Latest), account.address()).await.unwrap();

let deploy_tx = contract_factory.deploy(vec![], FieldElement::ZERO, true)
     // manually setting fee else estimate_fee will be called and it will fail
    // as contract is not declared yet (declared in the same block as deployment
    .max_fee(FieldElement::from_hex_be("0x1000000000").unwrap())
    // manually incrementing nonce else as both declare and deploy are in the same block
    // so automatic nonce calculation will fail
   .nonce(nonce + FieldElement::ONE);

````

Reply:
I think it's a more common usage of the builder pattern

Reply:
And I check your other usages of this `contract_factory`, maybe it can be done in a similar way there too

Reply:
yep, makes sense! fixed